posthog/management/commands/migrate_retention_show_mean.py

```diff
@@ -0,0 +1,67 @@
+from django.core.management.base import BaseCommand
+from django.db import transaction
+
+from posthog.models import Insight
+
+
+def migrate_show_mean_from_boolean_to_string(batch_size: int, live_run: bool = False) -> None:
+    """
+    Migrate the showMean boolean field to meanRetentionCalculation string field in retention insights.
+    """
+    retention_insights = Insight.objects.filter(
+        deleted=False,
```

Comment:
You don't need to add `deleted=False`, as that's the [default for the InsightsManager](https://github.com/PostHog/posthog/blob/a107d19f20f43adcb40f59b2d483cc81a970abaa/posthog/models/insight.py#L21-L23). I'd argue that we want to migrate delete insights as well though, and thus we should use `objects_including_soft_deleted`.